{
  for (  MembershipAction.JoinCallback callback : callbacks) {
    try {
      callback.onFailure(e);
    }
 catch (    Exception inner) {
      logger.error(new ParameterizedMessage("error handling task failure [{}]",e),inner);
    }
  }
}
